File: 2544/alternating-digit-sum.cpp
// LeetCode 2544. Alternating Digit Sum
// 每位數拆開、正、負交錯，加起來。
class Solution {
public:
    int alternateDigitSum(int n) {
        int sign = +1, ans = 0;
        for(char c : std::to_string(n)){
            ans += sign * (c-'0');
            sign *= -1;
        }
        return ans;
    }
};

File: 2544/alternating-digit-sum.py
# LeetCode 2544. Alternating Digit Sum
# 每位數拆開、正、負交錯，加起來。
class Solution:
    def alternateDigitSum(self, n: int) -> int:
        sign = +1
        ans = 0
        for c in str(n):
            ans += sign * int(c)  # 依「正負」號「加起來
            sign *= -1  # 正、負交錯
        return ans
